src/lib/utils/generalError.js

```diff
@@ -0,0 +1 @@
+export default errorCode => `Sorry, Some unexpected error (E${errorCode}) happened, please try again`
```

Comment:
1. I moved all error utils onto logger/exceptions.js Plese adjust all imports
2. use decorate() util to add code/name to the exception. Don't report the code with custom log payload. Also this utill will return 'Sorry some unexpected...' string you could show in the dialog to the endusers:

```javascript
import { decorate, ExceptionCode } from 'path/to/logger/exceptions'

const { E1, E2, ... } = ExceptionCode

// ... 
 const { message } = exception
 const uiMessage = decorate(exception, E1)

 // you don't  need to specify the code in the object param - decorate will set exception.code = 1 and exception.name = 'E1'
 log.error('evaluateFRValidity failed', message, exception, { dialogShown: true })
 showErrorDialog(uiMessage(1), '', { // ...
```

3. i see we still have incorrect error logging (e.g. evaluateFRValidity failed - no error object was logged just after the message). Please re check the app very carefully, find & fix all the similar cases